### DISCUSSION TITLE
PipeWriter and writing untyped data

### DISCUSSION THREAD

**Discussion by georgiosd** [JoshClose/CsvHelper#1965] [General]
Hey!

Congrats on writing the go-to CSV library in .NET :)

Two questions on your roadmap:

- Any intentions to support `PipeWriter`?

- Any intentions to support writing untyped data? (e.g. string array)

**Comment by JoshClose** [comment]
> Any intentions to support PipeWriter?

Not at this time no. What advantages would it have? Would using the `PipeWriter.AsStream` method work just as well?

> Any intentions to support writing untyped data? (e.g. string array)

You can get the row as a `string[]` from the parser. If you're using `CsvReader` you can get `reader.Parser.Record` or if using the `CsvParser` directly, `parser.Record`.

**Reply by georgiosd** [reply]
I think it's supposed to be faster. Will it matter for this? Maybe not.

Re untyped data, I was asking about *writing*, not reading.

**Reply by JoshClose** [reply]
> I think it's supposed to be faster. Will it matter for this? Maybe not.

I haven't looked into it for writing. Writing definitely isn't optimized.

> Re untyped data, I was asking about writing, not reading.

Probably not. It's easy enough to do it by hand. 

```cs
foreach (var record in records)
{
    foreach (var field in record)
    {
        csv.WriteField(field);
    }
    csv.NextRecord();
}
```

**Reply by georgiosd** [reply]
Ah, I didn't know you could do this! Thanks!
